server/aws-lsp-codewhisperer/src/language-server/codeWhispererService.ts

```diff
@@ -61,6 +61,13 @@ export abstract class CodeWhispererServiceBase {
         AWS.config.update(awsConfig)
     }
 
+    /**
+     * Updates Service Client options after client was instantiated.
+     */
+    public updateClientConfig(options: ConfigurationOptions) {
```

Comment:
Do we need this? AFAIK `updateAwsConfiguration` would apply globally so we can utilise that instead

Reply:
IDK and I didn't test that, that's why I created this specific implementation

Reply:
Thinking about this a bit more, I'd avoid setting global AWS configs from individual Language Server implementation, if it has side effects on other servers. This may result in unexpected result, where one server overrides settings in others servers

Reply:
Will different servers have different user agent info?

Reply:
Authors are free to set it according to their service expectations. Since SDK clients are part of server codebase, they can choose to any value and we can't/shouldn't enforce it